autocertkit/tests/xenapi_mock.py

```diff
@@ -254,36 +266,66 @@ class XenapiMock(mock.Mock):
     This only replicate all required xenapi component as properties.
     """
 
+    def __init__(self, session):
+        super(XenapiMock, self).__init__()
+        self.__session = session
+        self.__xenapiPool = XenapiPoolMock(self)
+        self.__xenapiHost = XenapiHostMock(self)
+        self.__xenapiHostMetrics = XenapiHostMetricsMock(self)
+        self.__xenapiPif = XenapiPIFMock(self)
+        self.__xenapiNetwork = XenapiNetworkMock(self)
+        self.__xenapiBond = XenapiBondMock(self)
+        self.__xenapiVm = XenapiVMMock(self)
+
+    @property
+    def session(self):
+        return self.__session
+
     @property
     def pool(self):
-        return XenapiPoolMock()
+        return self.__xenapiPool
 
     @property
     def host(self):
-        return XenapiHostMock()
+        return self.__xenapiHost
 
     @property
     def host_metrics(self):
-        return XenapiHostMetricsMock()
+        return self.__xenapiHostMetrics
 
     @property
     def PIF(self):
-        return XenapiPIFMock()
+        return self.__xenapiPif
 
     @property
     def network(self):
-        return XenapiPoolMock()
+        return self.__xenapiNetwork
 
     @property
     def bond(self):
-        return XenapiBondMock()
+        return self.__xenapiBond
 
     @property
     def VM(self):
-        return XenapiVMMock()
+        return self.__xenapiVm
 
 
-class XenapiNetworkMock(mock.Mock):
+class _XenapiSubclassMock(mock.Mock):
```

Comment:
name _XenapiSubclassMock does not represent what it is. I'd suggest 'XenapiMockBase'. This is basically mocking session.xenapi... api sets.